### ISSUE TITLE
Qualifying (non-bulitin) <Erased> DU's do not discriminate properly

### ISSUE THREAD

**Issue by thelazydogsback** [fable-compiler/Fable#736]
For example, in:

```
[<Erase>] type DU = | Int of int | Str of string 
let strify = function
  | Int i -> "i:" + string i 
  | Str s -> "s: " + s
```
the output looks like this:
```
export function strify(_arg1) {
  if (true) {
    return "s: " + _arg1;
  } else {
    return "i:" + String(_arg1);
  }
}
```
Expected code (as with built-in U2, U#, etc.):

```
export function strify(x) {
  if (typeof x === "string") {
    return "s: " + x;
  } else {
    return "i:" + String(x);
  }
}

```

**Comment by alfonsogarciacaro** [comment]
fable-compiler 0.7.48 has been released with a fix. Could you please give it a try?

**Comment by thelazydogsback** [comment]
Re:testing -- I've just been using fable.io/repl as my playground at the moment. (Which says Fable 1.0.0 but must be some previous 0.x version?)

**Comment by alfonsogarciacaro** [comment]
The REPL actually works with what will be the next Fable major version (`narumi` branch) so you may notice some differences. I'm working with both versions in parallel, in this case I've put the change in the released version (0.7.x) but it'll be propagated to the REPL the next time I update it :)
